client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeleteRoleMappingRequest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.common.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Request object to delete a role mapping.
+ */
+public final class DeleteRoleMappingRequest implements Validatable {
+    private final String name;
+    private final RefreshPolicy refreshPolicy;
+
+    public DeleteRoleMappingRequest(final String name, @Nullable final RefreshPolicy refreshPolicy) {
+        this.name = Objects.requireNonNull(name, "role-mapping name is missing");
+        this.refreshPolicy = (refreshPolicy == null) ? RefreshPolicy.getDefault() : refreshPolicy;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public RefreshPolicy getRefreshPolicy() {
+        return refreshPolicy;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Do we need to implement `hashCode()` and `equals()` ?

Reply:
No strong opinions here. It was more of a question than a comment . @jaymode WDYT ?

Reply:
Mental note that we'd need to revisit our already merged PRs to add these and tests if we decide on it.